packages/design-picker/src/components/index.tsx

```diff
@@ -119,22 +126,29 @@ const DesignButton: React.FC< DesignButtonProps > = ( {
 		let text: any = __( 'Free' );
 
 		if ( design.is_premium ) {
-			text = shouldUpgrade ? (
-				<Button
-					isLink={ true }
-					className="design-picker__button-link"
-					onClick={ ( e: any ) => {
-						e.stopPropagation();
-						onCheckout?.();
-					} }
-				>
-					{ 'en' === locale || hasTranslation( 'Included in WordPress.com Premium' )
-						? __( 'Included in WordPress.com Premium' )
-						: __( 'Upgrade to Premium' ) }
-				</Button>
-			) : (
-				__( 'Included in your plan' )
-			);
+			if ( shouldUpgrade ) {
```

Comment:
This conditional felt a little convoluted to me, do you think it's clearer like this?:

```
text = __( 'Included in your plan' );

if ( shouldUpgrade ) {
   text = <Button ...
}
if ( hasPurchasedTheme ) {
    text = __( 'Purchased on an annual subscription' );
}
```
This is just a suggestion, do as you feel it's better.

Reply:
I flatted into one level, with each condition being explicit and having no default. Take a look

Reply:
yep, much better now. Thanks!